src/index.js

```diff
@@ -0,0 +1,160 @@
+    updateTemp(temperature);
+    
+    // ------------- wave 3 naming the city ------------------
+    // Realtime Text City info will come from input value of cityInput
+    const updateCityName = () => {
+        cityName.textContent = cityInput.value;
+    }
+    
+    // event listener to listen for input
+    cityInput.addEventListener("input", updateCityName);
+    
+    // ------------- wave 5 selecting sky ------------------
+    result.textContent = "🌦 🌈  ☁️☁️☁️  ❄️ 🌨 ☁️"
+    selectSky.addEventListener("change", (event) => {
+        let newSky = event.target.value
+        if (newSky == 'sunny') {
+            result.textContent = "☁️ ☁️ ☁️ ☀️ ☁️ ☁️"
+        } else if (newSky == 'cloudy'){
+            result.textContent = "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️"
+        } else if (newSky == 'rainy'){
+            result.textContent = "🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧"
+        } else if (newSky == 'snowy'){
+            result.textContent = "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨"
+        } else {
+            result.textContent = "🧨 🧨 🧨 🧨 🧨 🧨"
+        }
+    })
+    
+    // -------- wave 6    reset button event listener ---------
+    cityName.textContent = state.city
+    resetButton.addEventListener("click", function () {
+        cityInput.value = state.city; 
+        cityName.textContent = state.city;
+        updateTemp("65")
+    } )
+
+    // -------- wave 4    calls API  ---------
+    const findLatitudeAndLongitude = async () => {
+        // let latitude, longitude;
+        await axios.get('http://127.0.0.1:5000/location',
+        {
+            params: {
+                q:cityInput.value,
+                format: 'json'
+            }
+        })
+        .then( (response) => {
+            state.lat= response.data[0].lat;
+            state.lon= response.data[0].lon;
+        })
+        .catch( (error) => {
+            console.log('error in findLatitudeAndLongitude!');
+        });
+        return {
+            cityLat: state.lat,
+            cityLon: state.lon
+        }
+    }
+
+    const findTemp = async (latitude, longitude) => {
+        await axios.get('http://127.0.0.1:5000/weather',
+        {
+            params: {
+                format: 'json',
+                lat: latitude,
+                lon: longitude
+            }
+        })
+        .then( (response) => {
+            temperature = response.data.main.temp;
+            return temperature;
+        })
+        .catch( (error) => {
+            console.log('error in findLocation!');
+        });
+    }
+
+    const findWeather = async () => {
+        const cityCoordinates = await findLatitudeAndLongitude();
+        await findTemp(cityCoordinates.cityLat, cityCoordinates.cityLon);
+    }
+
+// -------------------    search button    ---------------------
+    searchButton.addEventListener("click", async (event) => {
+        await findWeather();
+        convert()
+        temperature = Math.round(temperature)
+        updateTemp(temperature)
+
+    });
+
```

Comment:
Awesome job everyone! You really ate this one up! Please feel free to reach out to me if you have any questions about the comments I left or if you want to discuss anything in greater detail!  ⭐️